### PR TITLE
Add e2e support for aten logical or/and/xor/not ops

### PR DESCRIPTION
**Pull request by li-plus**
This PR supported lowering `AtenLogicalOrOp` to MHLO, and added `AtenLogicalAndOp`, `AtenLogicalXorOp` and `AtenLogicalNotOp` for both LINALG and MHLO backends.

### CODE REVIEW

**Review by tanyokwok — APPROVED** [review]
LGTM

File: lib/Conversion/TorchToMhlo/Basic.cpp [hunk 1]
```diff
@@ -1389,6 +1425,16 @@ void mlir::torch::torch_to_mhlo::populateBasicOpPatternsAndLegality(
...
 #undef INSERT_BINARY_COMPARE_PATTERN
 
+#define INSERT_BINARY_LOGICAL_PATTERN(AtenOp, ChloOp)                          \
+  target.addIllegalOp<AtenOp>();                                               \
+  patterns.add<ConvertAtenLogicalBinaryOp<AtenOp, ChloOp>>(typeConverter,      \
+                                                           context)
+
+  INSERT_BINARY_LOGICAL_PATTERN(AtenLogicalOrOp, chlo::BroadcastOrOp);
+  INSERT_BINARY_LOGICAL_PATTERN(AtenLogicalAndOp, chlo::BroadcastAndOp);
+  INSERT_BINARY_LOGICAL_PATTERN(AtenLogicalXorOp, chlo::BroadcastXorOp);
```
Comment (ramiro050) [inline lib/Conversion/TorchToMhlo/Basic.cpp:1435]: Is there a reason these changes don't pass the e2e tests?

**Comment by ashay** [comment]
@li-plus The E2E tests introduced in this commit seem to be non-deterministically failing in the post-merge build.  Here is an [issue](https://github.com/llvm/torch-mlir/issues/1758) to track the problem and PR https://github.com/llvm/torch-mlir/pull/1757 reverts this change so that subsequent PRs are unblocked.  Could you look into the failure please?  Thanks!

**Comment by li-plus** [comment]
@ashay Okay, I have reproduced the error locally and submitted a fix in #1761.

**Pull request merged**